src/lib.rs

```diff
@@ -0,0 +1,181 @@
+    start_y: f32,
+    alive: bool,
+}
+impl GameCell {
+    pub fn new(start_x: f32, start_y: f32, alive: bool) -> Self {
+        GameCell {
+            start_x,
+            start_y,
+            alive,
+        }
+    }
+    // Initial function to calculate whether two cells are adjacent -- TODO make smaller
+    pub fn is_adjacent(&self, cell: &GameCell) -> bool {
+        // This is Pythagoras which finds the distance from a cell's start
+        // point to another cell's start point that is diagonally placed
+        let max_allowed_distance = (DEFAULT_CELL_LEN.powi(2) * 2.0).sqrt();
+        // Calculates the distance between the two cells starting points
+        let distance = ((cell.x() - self.x()).powi(2) + (cell.y() - self.y()).powi(2)).sqrt();
+        // Distance can either be the max distance or the default edge len if the adjacent cell is not diagonally placed
+        distance.eq(&DEFAULT_CELL_LEN) || distance.eq(&max_allowed_distance)
+    }
+
+    pub fn is_alive(&self) -> bool {
+        self.alive
+    }
+    pub fn x(&self) -> f32 {
+        self.start_x
+    }
+    pub fn y(&self) -> f32 {
+        self.start_y
+    }
+
+    pub fn set_alive(&mut self, alive: bool) {
+        self.alive = alive;
+    }
+}
+
+// Function to generate the initial grid (none of the cells alive before the game starts)
+pub fn generate_cells(screen_width: f32, screen_height: f32) -> Vec<GameCell> {
+    // Calculate how many cells there will be based on the screen area and the default cell area
+    let cells_amount = ((screen_height * screen_width) / DEFAULT_CELL_AREA) as i32;
+    let mut offset_x = 0.;
+    let mut offset_y = 0.;
+    let mut cells: Vec<GameCell> = Vec::new();
+
+    for _ in 0..cells_amount {
+        let new_cell = GameCell {
+            start_x: offset_x,
+            start_y: offset_y,
+            alive: false,
+        };
+        cells.push(new_cell);
+
+        if offset_x == screen_width - DEFAULT_CELL_LEN {
+            offset_x = 0.;
+            offset_y += DEFAULT_CELL_LEN;
+        } else {
+            offset_x += DEFAULT_CELL_LEN;
+        }
+    }
+    cells
+}
+
+// Function to update current cells state (basically if they remain alive or not)
+pub fn update_cells(cells: &mut Vec<GameCell>) {
+    let cells_reference = cells.clone();
+    for s in cells {
+        // Obtain previously alive neighbours
+        let neighbors_amount = cells_reference
+            .iter()
+            .filter(|x| s.is_adjacent(x) && x.is_alive())
+            .count();
+        if s.is_alive() {
+            match neighbors_amount {
+                0..2 => s.alive = false, // Less than two alive neighbours -> cell dies
+                2..=3 => s.alive = true, // Two or three alive neighbours -> cell lives
+                4.. => s.alive = false,  // More than three alive neighbours -> cell dies
+            }
+        } else if !s.is_alive() && neighbors_amount == 3 {
+            s.alive = true;
+        }
```

Comment:
Tip: code can be simplified by doing a tuple `match`:
```suggestion
        match (s.is_alive(), neighbors_amount) {
            (true, 2..=3) => {} // A live cell with two or three neighbours is kept alive
            (false, 3) => s.alive = true, // A dead cell with three neighbours is revived
            _ => s.alive = false, // Any other case (dead or alive) cell will be dead
        };
```

Reply:
done